main.go

```diff
@@ -126,73 +109,251 @@ func (e *periodicExporter) Collect(ch chan<- prometheus.Metric) {
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_resource_mem_total_bytes", float64(stats.MemoryTotal), "Available bytes of memory available as resource")
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_resource_mem_used_bytes", float64(stats.MemoryUsed), "Bytes of memory resources in use")
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_resource_mem_used_percentage", float64(stats.MemoryPercentage), "Percentage of memory resources in use")
+
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_executors_registering", float64(stats.ExecutorsRegistering), "Amount of executors that are registering")
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_executors_running", float64(stats.ExecutorsRunning), "Amount of executors that are running")
+	e.counter(metricsChan, labels, labelValues, "mesos_slave_executors_terminated_total", float64(stats.ExecutorsTerminated), "Total of terminated executors")
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_executors_terminating", float64(stats.ExecutorsTerminating), "Amount of executors that are terminating")
+
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_frameworks_active", float64(stats.FrameworksActive), "Amount of active frameworks")
+	e.counter(metricsChan, labels, labelValues, "mesos_slave_invalid_framework_messages_total", float64(stats.InvalidFrameworkMessages), "Total of invalid framework messages")
+	e.counter(metricsChan, labels, labelValues, "mesos_slave_invalid_status_updates_total", float64(stats.InvalidStatusUpdates), "Total of invalid status updates")
+
+	e.counter(metricsChan, labels, labelValues, "mesos_slave_recovery_errors_total", float64(stats.RecoveryErrors), "Amount of errors during slave recovery")
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_registered", float64(stats.Registered), "Is slave registered with the master")
+
+	e.counter(metricsChan, labels, labelValues, "mesos_slave_tasks_failed_total", float64(stats.TasksFailed), "Total of failed tasks")
+	e.counter(metricsChan, labels, labelValues, "mesos_slave_tasks_finished_total", float64(stats.TasksFinished), "Total of finished tasks")
+	e.counter(metricsChan, labels, labelValues, "mesos_slave_tasks_killed_total", float64(stats.TasksKilled), "Total of killed tasks")
+	e.counter(metricsChan, labels, labelValues, "mesos_slave_tasks_lost_total", float64(stats.TasksLost), "Total of lost tasks")
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_tasks_running", float64(stats.TasksRunning), "Amount of running tasks")
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_tasks_staging", float64(stats.TasksStaging), "Amount of staging tasks")
+	e.gauge(metricsChan, labels, labelValues, "mesos_slave_tasks_starting", float64(stats.TasksStarting), "Amount of starting tasks")
+
+	e.counter(metricsChan, labels, labelValues, "mesos_slave_uptime_seconds", stats.UptimeSecs, "Slave uptime in seconds")
```

Comment:
I don't suppose there's a start time available rather than an uptime? If uptime gets stuck, you won't notice.